src/conn/conn_api.c

```diff
@@ -1067,11 +1067,9 @@ __conn_close(WT_CONNECTION *wt_conn, const char *config)
      * After the async and LSM threads have exited, we won't open more files for the application.
      * However, the sweep server is still running and it can close file handles at the same time the
      * final checkpoint is reviewing open data handles (forcing checkpoint to reopen handles). Shut
-     * down the sweep server and then flag the system should not open anything new.
+     * down the sweep server.
      */
     WT_TRET(__wt_sweep_destroy(session));
-    F_SET(conn, WT_CONN_CLOSING_NO_MORE_OPENS);
```

Comment:
I think it's important that we can assert this before the final checkpoint is completed, otherwise changes can be made that won't be reflected at shutdown.

Reply:
@agorrod,
> Is it possible to only do the cursor open/close dance when the no-more-opens flag isn't set? Or are there cases where eviction has work to do at that stage which involves the history store?

I don't see how we could avoid the obvious race where we test the flag and then close is called.

> I think it's important that we can assert this before the final checkpoint is completed, otherwise changes can be made that won't be reflected at shutdown.

Maybe we should swap out the `WT_CONN_CLOSING_NO_MORE_OPENS` flag for a no more writes flag? After all, it's not the open that we care about. Once the final checkpoint completes, nothing gets written or we assert. Would that be a good solution?